docs/contributing.md

```diff
@@ -117,31 +117,137 @@ fetch master and create these branches for you:
 
 ### 4. Commit your changes
 
-Commit your change. The commit command must include a specific message format:
+Commit your changes using the Conventional Commits format. All commits must follow the format described in the [Conventional Commits](#conventional-commits) section below and include proper git trailers for issue tracking.
 
+**Basic commit format:**
 ```bash
-git commit -m "<component>: <change_message> #<issue number>"
+git commit -m "<type>(<scope>): <description>" --trailer "Closes: #<issue_number>"
 ```
 
-Valid component names are listed in the [__label
-list__](https://github.com/rucio/rucio/labels) and are usually specified on the
-issue of the change.
+**Example:**
+```bash
+git commit -m "feat(Core): Add new rule evaluation engine" --trailer "Closes: #1234"
+```
 
 Add additional explanations to the body of the commit, such as motivation for
 certain decisions and background information. [Here are some general rules.](https://cbea.ms/git-commit/).
 
-If you add a [__github-recognised
-keyword__](https://help.github.com/articles/closing-issues-using-keywords/) then
-the associated issue can be closed automatically once the pull request is
-merged, e.g.:
+Using multiple commits is allowed as long as they achieve an independent,
+well-defined, change and are well-described. Otherwise multiple commits should
+be squashed.
+
+### **Conventional Commits**
```

Comment:
I think this should be one header level lower (`####`) since it's within the `### 4. Commit your changes` step